src/thumbnailer_slope_optim.cc

```diff
@@ -250,4 +267,80 @@ Thumbnailer::Status Thumbnailer::LossyEncodeNoSlopeOptim(
   return (webp_data->size > 0) ? kOk : kByteBudgetError;
 }
 
+Thumbnailer::Status Thumbnailer::ExtraLossyEncode(WebPData* const webp_data) {
+  // Encode frames following the ascending order of slope between frame's
+  // current quality and quality 100.
+  std::vector<std::pair<float, int>> encoding_order;
+  for (int i = 0; i < frames_.size(); ++i)
+    if (!frames_[i].near_lossless) {
+      float slope;
+      CHECK_THUMBNAILER_STATUS(
+          ComputeSlope(i, frames_[i].final_quality, 100, &slope));
+      encoding_order.push_back(std::make_pair(slope, i));
+    }
+  std::sort(encoding_order.begin(), encoding_order.end(),
+            [](const std::pair<float, int>& x, const std::pair<float, int>& y) {
+              return x.first < y.first;
+            });
+
+  WebPData new_webp_data;
+  WebPDataInit(&new_webp_data);
+  int num_remaining_frames = encoding_order.size();
```

Comment:
`num_remaining_frames` is equal to `encoding_order.size()` all the time so it could be removed.